### DISCUSSION TITLE
zh/guide/drivers/onedrive

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#34] [Announcements]
# zh/guide/drivers/onedrive

AList v3的文档

https://alist.nn.ci/zh/guide/drivers/onedrive.html

**Comment by buduan** [comment]
sharepoint站点地址是?

**Reply by qian9008** [reply]
在参数获取页点自动获取

**Reply by Java2Boy** [reply]
输入站点URL，提示错误

site_id: 获取中

error: Invalid hostname for this tenancy

**Reply by commandoccz** [reply]
说了等于没说，参数获取页面自动获取？？？

**Reply by commandoccz** [reply]
找了一圈，搞明白了，所谓sharepoint站点，是指在https://portal.office.com/里面的sharepoint这个应用，点击创建网站，选择团队私有并命名创建一个网站，此网站（例如：https://a1p6.sharepoint.com/sites/alist-engineer?market=zh-CN）即为 https://tool.nn.ci/onedrive/request中需要填入的site_url，此site_url参数跟配置文件中的alist网站域名不是一回事

**Comment by qian9008** [comment]
sharepoint站点填后 没有效果.. 是需要单独挂载吧。

**Reply by qian9008** [reply]
解决了 在是否sp站点 选上就是sp站点了

**Comment by ioliu** [comment]
Failed init storage: AADSTS9002313: Invalid request. Request is malformed or invalid. Trace ID: cbd3d527-408f-4e61-99b2-e3329971a500 Correlation ID: a59cecc5-d9f2-4224-88ac-e1ac079b7bb0 Timestamp: 2022-10-10 14:30:02Z


这种错误是什么情况，有大佬知道吗，求解答

**Reply by mojinxi** [reply]
刷新令牌一直在获取中

**Reply by ioliu** [reply]
解决了 应该是教育版权限的问题 我账号改为教育管理员以后就正常添加了

**Reply by zhong-honglin** [reply]
啊？还能改成管理员的吗

**Reply by ouclei** [reply]
这怎么改啊

**Comment by 839781960** [comment]
首先打开 https://tool.nn.ci/onedrive/request-------
1.打开后点创建应用--出现悬浮窗口显示需要交互，重新登录提示下面错误，无法登录。
所选的用户帐户在租户“Microsoft Services”中不存在，且无法访问该租户中的应用程序“18ed3507-a475-4ccb-b669-d66bc9f2a36e”。需要先将该帐户添加为该租户的外部用户。请使用其他帐户。

**Reply by q8y5i** [reply]
我也同样出现了这个问题，找了一晚上答案，可能是个人账号与工作或学校账号的区别，使用这种方式需要是组织账号。

不知道我的理解正确吗？！因为微软没有注册组织账号的方式，只能购买商业版软件方式注册，无法验证。

另外我好奇评论中的人都是使用的教育版并且有管理员权限吗？应该是只有组织账户的管理者才能修改这些东西吧~？

**Reply by q8y5i** [reply]
找到解决方法了 需要去微软云去免费注册 alist用起来还是很丝滑的

**Comment by Hewgy** [comment]
Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path
请问这个错误是我的授权没成功吗

**Reply by Hewgy** [reply]
找到错误所在了,输入路径的时候我不小心加了个空格:(

**Reply by ZYLeong95** [reply]
请问你是在输入什么路径的时候加了个空格？我只是填了挂载路径，添加的时候也出现了跟你一样的错误提示

**Reply by Hewgy** [reply]
我是在挂载路径里面添加了空格之后出现的错误

**Comment by luowenyu001** [comment]
加载onedrive文件时一直转圈，然后过一段时间就显示network error，是云服务器

**Reply by luowenyu001** [reply]
然后有了一些文件之后点进去显示 failed to load contenct

**Comment by zhang370521** [comment]
这是怎么了"Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path"

**Comment by Hope-L** [comment]
Key: 'Storage.MountPath' Error:Field validation for 'MountPath' failed on the 'required' tag

提示这个怎么办

**Reply by Hope-L** [reply]
重新填写提交后又出现了Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path
检查了没有多加空格，这个要怎么办？

**Comment by Comvertk** [comment]
Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path
请问一下这个怎么解决？感谢xdm回复

**Comment by XHLiang0** [comment]
failed get objs: failed to list objs: Resource not found for the segment 'root:E'.
这是啥原因？

**Comment by MrPPPPenguin** [comment]
Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path
怎么办

**Reply by makessr** [reply]
已经有相同名称的文件夹名称了，需要重新打开存储删除，然后再进行添加。

**Comment by makessr** [comment]
碰到一个很奇怪的问题，添加onedrive后，上传了文件不显示出来，把挂载的onedrive删掉重新添加也是一样不显示出来。有谁遇到过吗？？

**Reply by KuuBear** [reply]
请问解决了吗，我也是同样的问题

**Comment by cckunkun** [comment]
是网站问题吗？现在挂载onedrive，tool网址刷新令牌一直出不来

**Reply by bearduncle520** [reply]
一样

**Comment by yqianning** [comment]
注册账号倒是成功了，client_id 和 client_secret也拿到了，但是之后要登录onedrive账号，一直显示域名不存在

**Comment by ZY8080** [comment]
能够同时挂载两个不同的OneDrive账号吗？

**Reply by ZY8080** [reply]
错误提示Failed create storage in database: ERROR: duplicate key value violates unique constraint "x_storages_mount_path_key" (SQLSTATE 23505)

**Comment by codexss** [comment]
client_id 是新建应用 “概述” 页面的 ”应用程序(客户端) ID“ 
不是 “证书和密码” 的 “机密 ID”
填错了会报下面的错误：
```
unauthorized_client: The client does not exist or is not enabled for consumers.
```

**Reply by moyuwuhen601** [reply]
感谢

**Reply by Nicko8** [reply]
感谢，原文一步带过，没仔细看！

**Comment by niylin** [comment]
直接复制信息就挂载失败,备份的配置还原过来就没事是啥情况

**Comment by geekbb** [comment]
所有流程确保正常，挂载成功后提示：failed get objs: failed to list objs: Access Denied
求救

**Reply by geekbb** [reply]
是我自己的问题，我的onedrive超容被限制了，清理后现在好了！

**Comment by whao123yahoo** [comment]
获取site:id一直没有反应怎么回事

**Comment by zhong-honglin** [comment]
Using application 'alist' is currently not supported for your organization mails.ucas.ac.cn because it is in an unmanaged state. An administrator needs to claim ownership of the company by DNS validation of mails.ucas.ac.cn before the application alist can be provisioned这是为啥呀

**Comment by bebescom** [comment]
onedrive 打开提示：failed get objs: failed to list objs: Resource not found for the segment 'children'.

**Comment by yemaxul** [comment]
报错：failed get objs: failed to list objs: Get "https://graph.microsoft.com/v1.0/me/drive/root/children?$expand=thumbnails": EOF
之前的token能用，突然就不行了

**Comment by LxnChan** [comment]
请问一下 获取Onedrive刷新令牌 的这个后端可以自行部署（有开源）吗

**Reply by ddaodan** [reply]
可以，我以前在issue问过你可以看一下

**Comment by zhang370521** [comment]
不清楚，还有请问为什么向我提问




838400591
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "alist-org/docs"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月2日(星期四) 上午10:37
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [alist-org/docs] zh/guide/drivers/onedrive (Discussion #34)





 
请问一下 获取Onedrive刷新令牌 的这个后端可以自行部署（有开源）吗

**Comment by whao123yahoo** [comment]
就mountA一次，再mountB一次就行了呀

发自我的小米
在 yiwenleo ***@***.***>，2023年2月12日 05:21写道：

你好 请问我跟目录有 ABC三个文件夹 我只想mount A和B 不想显示C文件夹 如何操作

**Comment by jamyido** [comment]
SharePoint的Site ID调用似乎不可用？

**Comment by jamyido** [comment]
在 tool.nn迁移至alist.nn后就不可以使用了
其显示报错类似：Invalid hostname for this tenancy
我输入的是：https://huyihe.sharepoint.com/sites/yunjin
无论使用官方的cilent_id还是我自己的都不可用，不知道是我的问题还是alist官方后台的问题

**Reply by xhofe** [reply]
fixed in https://github.com/alist-org/docs/commit/f8100e23828d9bea36aabad77a3477dcfc134c8d

**Comment by ghost** [comment]
为什么获取刷新令牌一直转圈无法获取?

**Comment by qian9008** [comment]
Make sure the redirect URI sent in the request matches one added to your application in the Azure portal

**Reply by qian9008** [reply]
uri 填了老版本的就会出错。https://alist.nn.ci/tool/onedrive/callback 没问题。ok

**Comment by underlaw** [comment]
请问不能刷新新建的文件夹，或删除之类的操作，是什么原因。还有设置路径也不起作用是什么原因了

**Comment by KimmyLueng** [comment]
failed get objs: failed to list objs: You do not have access to create this personal site or you do not have a valid license  是什么问题

**Comment by Valeaf** [comment]
为什么用这种方式挂在OneDrive之后，能够正常打开挂在的文件夹，但是看不到Word保存在OneDrive里面的文件呢？

**Comment by Winyer** [comment]
新版本的分片大小是什么？是Onedrive的分块大小吗？建议最大60MB那个？

**Comment by woshichenghaibo** [comment]
谁知道这个东西是不是支持呃个人用户one drive的？因为不是所有人都有one drive business账户

**Reply by yiwenleo** [reply]
大哥 你这属于在凯迪拉克车友群里面说有的人没钱去洗浴中心一样

**Comment by Logosww** [comment]
报错 failed get objs: failed to list objs: Tenant does not have a SPO license. 是什么原因？

**Comment by papersnake** [comment]
我们无法完成你的请求
unauthorized_client: The client does not exist or is not enabled for consumers. If you are the application developer, configure a new application through the App Registrations in the Azure Portal at https://go.microsoft.com/fwlink/?linkid=2083908. 这是因为我的帐号没有权限吗？

**Comment by mrseawood** [comment]
请问一下 使用 默认提供的client 获取的 刷新令牌会不会过期呢
如果会过期的话 大概多久会过期
谢谢

**Comment by yubanmeiqin9048** [comment]
3.20.1版本，webdav策略选择「使用代理地址」或「本地代理」，在通过webdav拉取文件时都不使用设置的下载代理URL

**Comment by ahjdfeifei** [comment]
failed get objs: failed to list objs: User not found
是不是个人账户不行吗?

**Comment by ankey1989** [comment]
获取令牌出现以下错误是什么原因啊，全程照葫芦画瓢
unauthorized_client：客户端不存在或未为使用者启用。如果你是应用程序开发人员，请在 Azure 门户中通过 https://go.microsoft.com/fwlink/?linkid=2083908 处的应用注册配置新应用程序。

**Comment by jgc9527** [comment]
不能挂载到子用户的账号下吗

**Comment by relidaning** [comment]
请问， 我通过aria离线下载完成之后， 在alist将文件从本地传递到onedrive时异常， 在aria任务处提示提示Put “http://...”: write tcp [local_ipv6]:48690->[onedrive_domain_ipv6]:443: write: connection reset by peer, 这个是什么问题呢？

**Comment by bearduncle520** [comment]
提示 failed get objs: failed to list objs: The resource could not be found.
oneindex可以正常挂载，哎

**Comment by littlerm777** [comment]
安装完成后，点击onedrive文件夹出现：failed get objs: failed to list objs: Unable to retrieve user's mysite URL.请问是什么原因啊？谢谢

**Comment by LW030406** [comment]
为什么无法同步我上传的文件

**Comment by newcdn** [comment]
挂载完onedrive出现一个lost+found的文件夹，也删不掉，怎么办

**Reply by xuanxuan1231** [reply]
应该是系统自动生成的文件夹

**Comment by LW030406** [comment]
？取得 Android 版 Outlook

**Comment by buynonsense** [comment]
全部弄好后在alist中打开，显示failed get objs: failed to list objs: Item does not exist

**Reply by buynonsense** [reply]
已解决：我把alist中配置路径时写错了

**Comment by ljcbaby** [comment]
好像没法显示来自其他人共享的，添加到自己目录的文件夹

**Comment by lee-cq** [comment]
现在SharePrint是不是变成租户全局1T了？最近总是收到邮件说超了

**Comment by LW030406** [comment]
不知福啊获取Outlook for Android发件人: ***@***.*** ***@***.***> 代表 Lee CQ ***@***.***>发送时间: 2024年3月4日星期一 22:52收件人: docs ***@***.***>抄送: 2734708970 ***@***.***>; comment ***@***.***>主题: Re: [alist-org/docs] zh/guide/drivers/onedrive (Discussion #34)
现在SharePrint是不是变成租户全局1T了？最近总是收到邮件说超了

**Comment by LW030406** [comment]
不知道啊获取Outlook for Android发件人: ***@***.*** ***@***.***> 代表 Lee CQ ***@***.***>发送时间: 2024年3月4日星期一 22:52收件人: docs ***@***.***>抄送: 2734708970 ***@***.***>; comment ***@***.***>主题: Re: [alist-org/docs] zh/guide/drivers/onedrive (Discussion #34)
现在SharePrint是不是变成租户全局1T了？最近总是收到邮件说超了

**Comment by lianxyq** [comment]
我们无法完成你的请求unauthorized_client：客户端不存在或未为使用者启用。如果你是应用程序开发人员，请在 Azure 门户中配置新应用程序，https://go.microsoft.com/fwlink/?linkid=2083908。
成功了
ID是"应用程序(客户端) ID"

**Comment by John-881** [comment]
failed get objs: failed to list objs: Get "https://graph.microsoft.com/v1.0/me/drive/root/children?$top=5000&$expand=thumbnails($select=medium)&$select=id,name,size,lastModifiedDateTime,content.downloadUrl,file,parentReference": net/http: TLS handshake timeout
这个是什么情况

**Reply by John-881** [reply]
已解决：貌似是网络问题，等待一段时间刷新就可以了

**Comment by WongKwongYu** [comment]
无权开api，也无权关authenticator，是不是就不能用alist了？？为啥rclone没这问题

**Comment by Crazyboyjw** [comment]
注册应用提示在目录外部创建应用程序的功能已被弃用。你可通过加入 M365 开发人员计划或注册 Azure 来获取新目录。了解详细信息。
